src/interrupt.c

```diff
@@ -40,12 +40,14 @@ static void unknown_exception( int i, int code )
 
 	if(i==14) {
 		asm("mov %%cr2, %0" : "=r" (vaddr) );
-		if(pagetable_getmap(current->pagetable,vaddr,&paddr)) {
+        int data = vaddr < (int)current->brk;
+        int stack = vaddr > (int)current->brk && pagetable_getmap(current->pagetable,vaddr+PAGE_SIZE,&paddr);
```

Comment:
I'm not quite following the logic of this line.  If the page was validly mapped, we should have a page fault at all, right?